lib/jsdom/living/geometry/DOMMatrixReadOnly-impl.js

```diff
@@ -0,0 +1,519 @@
+      this._initialize3dMatrix(matrixInit);
+    } else {
+      throw new TypeError(`Invalid length for constructor init, expected 6 or 16, but received ${matrixInit.length}`);
+    }
+  }
+
+  parseDomString(domstring) {
+    const matrixFns = parseDOMStringToMatrixFunctions(domstring);
+    const baseMatrix = DOMMatrix.createImpl([]);
+
+    matrixFns.forEach(([fnName, fnArgs]) => {
+      switch (fnName) {
+        case "matrix3d":
+        case "matrix":
+          baseMatrix.multiplySelf(DOMMatrix.createImpl([fnArgs]));
+          break;
+        case "translate3d":
+        case "translate":
+        case "translateX":
+          baseMatrix.translateSelf(...fnArgs);
+          break;
+        case "translateY":
+          baseMatrix.translateSelf(0, fnArgs[0]);
+          break;
+        case "translateZ":
+          baseMatrix.translateSelf(0, 0, fnArgs[0]);
+          break;
+        case "scale3d":
+          baseMatrix.scaleSelf(...fnArgs);
+          break;
+        case "scale":
+          baseMatrix.scaleSelf(fnArgs[0], fnArgs[1], 1);
+          break;
+        case "scaleX":
+          baseMatrix.scaleSelf(fnArgs[0], 1, 1);
+          break;
+        case "scaleY":
+          baseMatrix.scaleSelf(1, fnArgs[0]);
+          break;
+        case "scaleZ":
+          baseMatrix.scaleSelf(1, 1, fnArgs[0]);
+          break;
+        case "rotate3d":
+          baseMatrix.rotateAxisAngleSelf(...fnArgs);
+          break;
+        case "rotate":
+        case "rotateZ":
+          baseMatrix.rotateAxisAngleSelf(0, 0, 1, fnArgs[0]);
+          break;
+        case "rotateX":
+          baseMatrix.rotateAxisAngleSelf(1, 0, 0, fnArgs[0]);
+          break;
+        case "rotateY":
+          baseMatrix.rotateAxisAngleSelf(0, 1, 0, fnArgs[0]);
+          break;
+        case "perspective":
+          baseMatrix.perspectiveSelf(...fnArgs);
+          break;
+        case "skew":
+          baseMatrix.skewSelf(...fnArgs);
+          break;
+        case "skewX":
+          baseMatrix.skewX(...fnArgs);
+          break;
+        case "skewY":
+          baseMatrix.skewY(...fnArgs);
+          break;
+      }
+    });
+    return arrayFromDOMMatrixInit(baseMatrix);
+  }
+
+  _initialize2dMatrix(init) {
+    matrix2dArrayProperties.forEach((prop, idx) => {
+      this[prop] = init[idx];
+    });
+
+    this.m33 = 1;
+    this.m44 = 1;
+    this._is2D = true;
+  }
+
+  _initialize3dMatrix(init) {
+    matrix3dArrayProperties.forEach((prop, idx) => {
+      this[prop] = init[idx];
+    });
+  }
+
+  _getCurrentMatrix() {
+    if (this.is2D) {
+      return matrix2dArrayProperties.map(prop => this[prop]);
+    }
+
+    return matrix3dArrayProperties.map(prop => this[prop]);
+  }
+
+  _updatePropsFromMatrix(matrix) {
+    matrix3dArrayProperties.forEach((prop, idx) => {
+      this[prop] = matrix[idx];
+    });
+  }
+
+  * [Symbol.iterator]() {
+    for (const i of matrix3dArrayProperties.map(prop => this[prop])) {
+      yield i;
+    }
+  }
```

Comment:
I don't see this being used anywhere. Let's just remove it.

Reply:
This is actually a helper for using `Array.from(DOMMatrix)` it makes writing some of the transform methods a lot cleaner

Reply:
Ah. In that case, I'd just call this function `_to1DArray`, and call `matrix._to1DArray()` explicitly.

If you insist on using `[Symbol.iterator]`, then you could just use an ordinary (non-generator) function:

```js
[Symbol.iterator]() {
  return matrix3dArrayProperties.map(prop => this[prop]).values();
}
```

Yet another way that's still slightly simpler is

```js
* [Symbol.iterator]() {
  yield* matrix3dArrayProperties.map(prop => this[prop]);
}
```

Reply:
No need for an iterator :) I'll change it

Reply:
No need for an iterator :) I'll change it to a plain old method